_release/bat/base_bat/base_bat_test.go

```diff
@@ -311,6 +326,160 @@ func TestDeleteAllInstances(t *testing.T) {
 	}
 }
 
+//Launch an instance created from an image with a customized cloud init
+//
+//Download cirros image, add it to the image service using a custom cloudinit file
+//create a workload, launch an instance of this workload and connect via SSH
+//using the custom user and password
+//
+//instance should accept connection with specified user and password and
+//reply with the user name
+func TestLaunchCustomInstance(t *testing.T) {
+	//creating a new cirros image
+	const name = "cirros-image"
+	ctx, cancelFunc := context.WithTimeout(context.Background(), standardTimeout)
+	defer cancelFunc()
+
+	// TODO:  The only options currently supported by the image service are
+	// ID and Name.  This code needs to be updated when the image service's
+	// support for meta data improves.
+	options := bat.ImageOptions{
+		Name: name,
+	}
+	//downaload cirros image
+	resp, err := http.Get("http://download.cirros-cloud.net/0.3.4/cirros-0.3.4-x86_64-disk.img")
```

Comment:
I'm just wondering why we download this image and whether this is a good idea.   Another alternative would be to check for an existing image with a known name and use this.  For example, Fedora Cloud Base 24-1.2.  If the image is not present you can just call t.Skip() and the test won't run.  This would make your test simpler.  There are other tests that verify that we can upload things to the image service.